persistence/sql/sqltest/base_basic.go

```diff
@@ -118,6 +118,245 @@ func SQLBasicTest(ass *assert.Assertions, store persistence.ProcessStore) {
 	describeProcess(ctx, ass, store, namespace, processId, xdbapi.COMPLETED)
 }
 
+func SQLProcessIdReusePolicyAllowIfPreviousExitAbnormally(ass *assert.Assertions, store persistence.ProcessStore) {
+	ctx := context.Background()
+	namespace := "test-ns"
+	processId := fmt.Sprintf("test-prcid-%v", time.Now().String())
+	input := createTestInput()
+
+	// Start the process and verify it started correctly.
+	startProcess(ctx, ass, store, namespace, processId, input)
+
+	// Try to start the process again and verify the behavior.
+	retryStartProcessForFailure(ctx, ass, store, namespace, processId, input)
+
+	// Describe the process.
+	describeProcess(ctx, ass, store, namespace, processId, xdbapi.RUNNING)
+
+	// stop the process with temerminated
+	terminateProcess(ctx, ass, store, namespace, processId)
+
+	// Describe the process, verify it's terminated
+	describeProcess(ctx, ass, store, namespace, processId, xdbapi.TERMINATED)
+
+	// start the process with allow if previous exit abnormally
+	startProcessWithAllowIfPreviousExitAbnormally(ctx, ass, store, namespace, processId, input)
+}
+
+func SQLProcessIdReusePolicyTerminateIfRunning(ass *assert.Assertions, store persistence.ProcessStore) {
+	ctx := context.Background()
+	namespace := "test-ns"
+	processId := fmt.Sprintf("test-prcid-%v", time.Now().String())
+	input := createTestInput()
+
+	// Start the process and verify it started correctly.
+	prcExeId := startProcess(ctx, ass, store, namespace, processId, input)
+
+	// Try to start the process again and verify the behavior.
+	retryStartProcessForFailure(ctx, ass, store, namespace, processId, input)
+
+	// Describe the process.
+	describeProcess(ctx, ass, store, namespace, processId, xdbapi.RUNNING)
+
+	prcExeID2 := startProcessWithTerminateIfRunningPolicy(ctx, ass, store, namespace, processId, input)
+
+	ass.NotEqual(prcExeId.String(), prcExeID2.String())
+}
+
+func SQLProcessIdReusePolicyDisallowReuseTest(ass *assert.Assertions, store persistence.ProcessStore) {
+	ctx := context.Background()
+	namespace := "test-ns"
+	processId := fmt.Sprintf("test-prcid-%v", time.Now().String())
+	input := createTestInput()
+
+	// Start the process and verify it started correctly.
+	prcExeId := startProcess(ctx, ass, store, namespace, processId, input)
+
+	// Try to start the process again and verify the behavior.
+	retryStartProcessForFailure(ctx, ass, store, namespace, processId, input)
+
+	// Describe the process.
+	describeProcess(ctx, ass, store, namespace, processId, xdbapi.RUNNING)
+
+	// Test waitUntil API execution
```

Comment:
you probably don't have to test the worker task code path. I like the way the above test cases doing it to be more straightfoward

Reply:
this is to make the process complete and then start again

Reply:
Ohh I see . Makes sense to me